Jenkinsfile

```diff
@@ -58,11 +58,30 @@ def runPerformanceTests(String testsPath, String stancFlags = ""){
     """
 }
 
+def cleanCheckout() {
+    retry(3) {
+        checkout([
+            $class: 'GitSCM',
+            branches: scm.branches,
+            extensions: [[$class: 'CloneOption', noTags: false]],
+            userRemoteConfigs: scm.userRemoteConfigs,
+        ])
+    }
+    
+    sh 'git clean -xffd'
+}
+
 pipeline {
     agent none
     parameters {
         booleanParam(name:"skip_end_to_end", defaultValue: false, description:"Skip end-to-end tests ")
         booleanParam(name:"skip_compile_O1", defaultValue: false, description:"Skip compile tests that run with STANCFLAGS = --O1")
+        booleanParam(name:"skip_compile", defaultValue: false, description:"Skip compile tests")
+        booleanParam(name:"skip_math_func_expr", defaultValue: false, description:"Skip math functions expressions test")
+        booleanParam(name:"skip_ocaml_tests", defaultValue: false, description:"Skip ocaml tests")
+        booleanParam(name:"skip_code_formatting", defaultValue: false, description:"Skip code formatting")
+        booleanParam(name:"skip_build", defaultValue: false, description:"Skip code formatting")
+        booleanParam(name:"skip_verify_changes", defaultValue: false, description:"Skip verify changes")
```

Comment:
These 3 are fast and shouldn't be skippable, IMO. Verify changes also seems maybe redundant with the others?